packages/components/pages/f-checkout/src/store/checkout.module.js

```diff
@@ -412,39 +423,35 @@ export default {
          * @param {Object} payload - Parameter with the different configurations for the request.
          */
         getGeoLocation: async ({ commit, state }, {
-            url, postData, timeout, tenant
+            url, postData, timeout, tenant, cookies
         }) => {
             let addressCoords;
-
+            const addressDetails = state.address;
+            const shouldLoadAddressFromLocalStorage = addressService.shouldLoadAddressFromLocalStorage(tenant);
             const isAddressInLocalStorage = addressService.isAddressInLocalStorage();
 
-            if (isAddressInLocalStorage) {
-                const storedAddress = addressService.getAddressFromLocalStorage(tenant, false);
-
-                if (addressService.doesAddressInStorageAndFormMatch(storedAddress, state.address)) {
-                    addressCoords = storedAddress.Field1 && storedAddress.Field2 ? [storedAddress.Field2, storedAddress.Field1] : null;
-                    commit(UPDATE_GEO_LOCATION, addressCoords);
-                } else {
-                    commit(UPDATE_GEO_LOCATION, null);
-
-                    const addressDetails = state.address;
-
-                    window.localStorage.setItem('je-full-address-details', JSON.stringify({
-                        PostalCode: addressDetails.postcode,
-                        Line1: addressDetails.line1,
-                        Line2: addressDetails.line2,
-                        Line3: addressDetails.line3,
-                        Line4: addressDetails.line4,
-                        administrativeArea: addressDetails.administrativeArea,
-                        City: addressDetails.locality
-                    }));
+            if (shouldLoadAddressFromLocalStorage) {
```

Comment:
this is a lot of nested if statements

Reply:
We may improve the nested if case by combining the conditions from if clauses, and replace else to another condition. What do you think?

Reply:
well I feel like the work should be done in the address service so instead here you just have for example `addressService.getAddress` whether the address is returned from cookie or localstorage is the responsibility of the address service.

Reply:
this would avoid duplicating 
```
const storedAddress = addressService.getAddressFromLocalStorage(tenant, false);
addressCoords = getGeoLocationFromAddress(addressDetails, storedAddress);